### PR TITLE
refactor: replace char_u with char 20

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

File: src/nvim/drawline.c [hunk 1]
```diff
@@ -616,16 +616,14 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
   int c_extra = NUL;                  // extra chars, all the same
   int c_final = NUL;                  // final char, mandatory if set
   int extra_attr = 0;                 // attributes when n_extra != 0
-  static char_u *at_end_str = (char_u *)"";  // used for p_extra when displaying
-                                             // curwin->w_p_lcs_chars.eol at
-                                             // end-of-line
+  static char *at_end_str = "";       // used for p_extra when displaying curwin->w_p_lcs_chars.eol at end-of-line
```
Comment (zeertzjq) [inline src/nvim/drawline.c:619]:
```suggestion
  static char *at_end_str = "";  // used for p_extra when displaying curwin->w_p_lcs_chars.eol at end-of-line
```

File: src/nvim/ex_getln.c [hunk 2]
```diff
@@ -3964,18 +3964,16 @@ char *vim_strsave_fnameescape(const char *const fname, const int what)
...
-# define PATH_ESC_CHARS ((char_u *)" \t\n*?[{`$\\%#'\"|!<")
-# define SHELL_ESC_CHARS ((char_u *)" \t\n*?[{`$\\%#'\"|!<>();&")
-# define BUFFER_ESC_CHARS ((char_u *)" \t\n*?[`$\\%#'\"|!<")
-  char *p =
-    (char *)vim_strsave_escaped((const char_u *)fname,
-                                what == VSE_SHELL ? SHELL_ESC_CHARS
-                                : what == VSE_BUFFER ? BUFFER_ESC_CHARS : PATH_ESC_CHARS);
+# define PATH_ESC_CHARS (" \t\n*?[{`$\\%#'\"|!<")
+# define SHELL_ESC_CHARS (" \t\n*?[{`$\\%#'\"|!<>();&")
+# define BUFFER_ESC_CHARS (" \t\n*?[`$\\%#'\"|!<")
```
Comment (zeertzjq) [inline src/nvim/ex_getln.c:3969]:
```suggestion
# define PATH_ESC_CHARS " \t\n*?[{`$\\%#'\"|!<"
# define SHELL_ESC_CHARS " \t\n*?[{`$\\%#'\"|!<>();&"
# define BUFFER_ESC_CHARS " \t\n*?[`$\\%#'\"|!<"
```

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/drawline.c [hunk 3]
```diff
@@ -616,16 +616,14 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
   int c_extra = NUL;                  // extra chars, all the same
   int c_final = NUL;                  // final char, mandatory if set
   int extra_attr = 0;                 // attributes when n_extra != 0
-  static char_u *at_end_str = (char_u *)"";  // used for p_extra when displaying
-                                             // curwin->w_p_lcs_chars.eol at
-                                             // end-of-line
+  static char *at_end_str = "";  // used for p_extra when displaying curwin->w_p_lcs_chars.eol at end-of-line
```
Comment (zeertzjq) [inline src/nvim/drawline.c:619]: Hmm, this line is 109 character long. Why isn't there a lint failure?

Comment (dundargoc) [reply src/nvim/drawline.c:619]: I've removed it. I thought uncrustify took care of that. Apparently there's a separate `cmt_width` option that we're not using.

**Pull request merged**